### PR TITLE
Fix GDNative library resource loading.

### PR DESCRIPTION
**Pull request by Faless**
Store general properties in ConfigFile too when modifying them.
Additionally set config_file property as internal as it's not editable
from inspector. It also does not appear to get saved in a meaningful way
(saved as Object(ConfigFile, ...))

### CODE REVIEW

File: modules/gdnative/gdnative.cpp [hunk 1]
```diff
@@ -243,7 +243,7 @@ void GDNativeLibrary::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("set_symbol_prefix", "symbol_prefix"), &GDNativeLibrary::set_symbol_prefix);
 	ClassDB::bind_method(D_METHOD("set_reloadable", "reloadable"), &GDNativeLibrary::set_reloadable);
 
-	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "config_file", PROPERTY_HINT_RESOURCE_TYPE, "ConfigFile"), "set_config_file", "get_config_file");
+	ADD_PROPERTY(PropertyInfo(Variant::OBJECT, "config_file", PROPERTY_HINT_RESOURCE_TYPE, "ConfigFile", PROPERTY_USAGE_INTERNAL), "set_config_file", "get_config_file");
```
Comment (karroffel) [inline modules/gdnative/gdnative.cpp]: What exactly does `PROPERTY_USAGE_INTERNAL` do? This property should be assignable via script for example but not shown in the inspector or saved inside the `.tres` . Does this do that?

**Review by karroffel — APPROVED** [review]
Looks good apart from me not knowing the PropertyHint :sweat_smile:

File: modules/gdnative/gdnative.cpp (hunk 1 above)
Comment (Faless) [reply modules/gdnative/gdnative.cpp]: @karroffel you are right, `INTERNAL` is not the right one, it hides the property from the docs (leaves setters and getters).
The right value seems to be `0` (and it's used like that around the engine).
Maybe we should have a `PROPERTY_USAGE_NONE = 0` in `PropertyUsageFlags`?
@reduz

**Review by karroffel — APPROVED** [review]
Seems good!

**Comment by akien-mga** [comment]
Is this safe to cherrypick for 3.1?

**Comment by karroffel** [comment]
Seems safe, it only keeps the data in the `ConfigFile` in sync.

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**